spark_client/services.py

```diff
@@ -0,0 +1,845 @@
+
+    def _retrieve_account_configurations(
+        self, name: str, namespace: str
+    ) -> PropertyFile:
+        secret_name = self._get_secret_name(name)
+
+        try:
+            secret = self.kube_interface.get_secret(secret_name, namespace=namespace)[
+                "data"
+            ]
+        except Exception:
+            return PropertyFile.empty()
+
+        return PropertyFile(secret)
+
+    def _build_service_account_from_raw(self, metadata: Dict[str, Any]):
+        name = metadata["name"]
+        namespace = metadata["namespace"]
+        primary = self.PRIMARY_LABEL in metadata["labels"]
+
+        return ServiceAccount(
+            name=name,
+            namespace=namespace,
+            primary=primary,
+            api_server=self.kube_interface.api_server,
+            extra_confs=self._retrieve_account_configurations(name, namespace),
+        )
+
+    def set_primary(self, account_id: str) -> str:
+        """Set the primary account to the one related to the provided account id.
+
+        Args:
+            account_id: account id to be elected as new primary account
+        """
+
+        # Relabeling primary
+        primary_account = self.get_primary()
+
+        if primary_account is not None:
+            self.kube_interface.set_label(
+                "serviceaccount",
+                primary_account.name,
+                f"{self.PRIMARY_LABEL}-",
+                primary_account.namespace,
+            )
+            self.kube_interface.set_label(
+                "rolebinding",
+                f"{primary_account.name}-role",
+                f"{self.PRIMARY_LABEL}-",
+                primary_account.namespace,
+            )
+
+        service_account = self.get(account_id)
+
+        if service_account is None:
+            raise NoAccountFound(account_id)
+
+        self.kube_interface.set_label(
+            "serviceaccount",
+            service_account.name,
+            f"{self.PRIMARY_LABEL}=True",
+            service_account.namespace,
+        )
+        self.kube_interface.set_label(
+            "rolebinding",
+            f"{service_account.name}-role",
+            f"{self.PRIMARY_LABEL}=True",
+            service_account.namespace,
+        )
+
+        return account_id
+
+    def create(self, service_account: ServiceAccount) -> str:
+        """Create a new service account and return ids associated id.
+
+        Args:
+            service_account: ServiceAccount to be stored in the registry
+        """
+
+        rolebindingname = service_account.name + "-role"
+        roleaccess = "view"
+
+        self.kube_interface.create(
+            "serviceaccount", service_account.name, namespace=service_account.namespace
+        )
+        self.kube_interface.create(
+            "rolebinding",
+            rolebindingname,
+            namespace=service_account.namespace,
+            **{"role": roleaccess, "serviceaccount": service_account.id},
```

Comment:
we need a custom role with minimal fine grained permissions.

Reply:
Yes, let's address this properly in a separate task!